### PR TITLE
update on conflict with condition

### PR DESCRIPTION
**Pull request by zidaye**
close #734 
refer to：
https://www.postgresql.org/docs/current/sql-insert.html

### CODE REVIEW

**Comment by baoyachi** [comment]
LGTM

File: src/ast/mod.rs [hunk 1]
```diff
@@ -2655,7 +2655,16 @@ pub struct OnConflict {
...
+    DoUpdate(DoUpdate),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct DoUpdate {
+    /// Column assignments
+    pub assignments: Vec<Assignment>,
+    /// WHERE
+    pub selection: Option<Expr>,
```
Comment (alamb) [inline src/ast/mod.rs:2667]: My preference is to call this `where` to match the SQL more closely -- I will make this change in a follow on PR

**Review by alamb — APPROVED** [review]
Thank you @zidaye

**Comment by alamb** [comment]
Thanks again @zidaye

File: src/ast/mod.rs (hunk 1 above)
Comment (alamb) [reply src/ast/mod.rs:2667]: Actually, now I see the name `selection` is consistent with the rest of this crate (e.g.  https://github.com/sqlparser-rs/sqlparser-rs/blob/faf75b71613431e66ec22c82e3c9ad6157474458/src/ast/mod.rs#L1139)

Thank you

**Pull request merged**